engine/common/follower/engine.go

```diff
@@ -205,12 +205,14 @@ func (e *Engine) processQueuedBlocks(doneSignal <-chan struct{}) error {
 
 		msg, ok := e.pendingBlocks.Pop()
 		if ok {
-			in := msg.(flow.Slashable[messages.BlockProposal])
-			err := e.processBlockProposal(in.OriginID, in.Message)
-			if err != nil {
-				return fmt.Errorf("could not handle block proposal: %w", err)
+			batch := msg.(flow.Slashable[[]*messages.BlockProposal])
+			for _, block := range batch.Message {
+				err := e.processBlockProposal(batch.OriginID, block)
+				if err != nil {
+					return fmt.Errorf("could not handle block proposal: %w", err)
+				}
```

Comment:
_Question_ to make sure I understand the bigger picture: this code block will be changed later to properly process a batch in one go (provided the batch is a sequence of connected blocks)? Is that correct?

Reply:
Yes, precisely, I have left it this way so we don't break anything in the current implementation.